### PR TITLE
Add SBT versioning plugin to help with releases, and describe process

### PR DESCRIPTION
**Pull request by jypma**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -109,6 +165,7 @@ lazy val `ts-reaktive-java` = project
     // This project includes Java -> Scala bridge classes, so we do want the scala library.
     autoScalaLibrary := true
   )
+  .enablePlugins(GitVersioning)
```
Comment (vizog) [inline build.sbt]: would [this](https://stackoverflow.com/questions/24683687/how-to-enable-sbt-plugins-of-root-project-for-all-subprojects) help in enabling the plugin for all subprojects?

Comment (jypma) [reply build.sbt]: no, since that requires a `.scala` build file, which I don't think SBT 1.0 does anymore.

File: build.sbt [hunk 2]
```diff
@@ -8,10 +8,58 @@ val kamonVersion = "0.6.6"
...
+crossScalaVersions := Seq("2.11.11", "2.12.3")
+
+releaseProcess := Seq(
+  checkSnapshotDependencies,
+  inquireVersions,
+  setReleaseVersion,
+  runClean,
+  runTest, 
+  tagRelease,
+//  publishArtifacts,
```
Comment (alar17) [inline build.sbt]: do you want to keep this?

Comment (jypma) [reply build.sbt]: ah, yes that should go back in.

**Pull request merged**